src/storage/upgrade.rs

```diff
@@ -0,0 +1,117 @@
+use crate::errors::*;
+use crate::storage::schema::{self, SchemaVersion, LATEST_SCHEMA_VERSION};
+use crate::storage::Transaction;
+
+pub fn upgrade(tx: &mut Transaction) -> Result<()> {
+    let current_version = schema::current_schema_version(tx)?;
+
+    info!(
+        "Database schema has version {:?}, latest schema version is {}",
+        current_version, LATEST_SCHEMA_VERSION
+    );
```

Comment:
Even though `SchemaVersion` implements `Display`, `Option<SchemaVersion>` (the type of `current_version`) does not. So I used `{:?}` instead of `{}`, which means the log will not look good at all :)

The alternatives involved writing more code, which seemed overkill for a mere log. Let me know if you disagree.